scylla-macros/src/value_list.rs

```diff
@@ -0,0 +1,26 @@
+use proc_macro::TokenStream;
+use quote::quote;
+
+/// #[derive(ValueList)] allows to parse a struct as a list of values,
+/// which can be fed to the query directly.
+/// Works only on simple structs without generics etc
+pub fn value_list_derive(tokens_input: TokenStream) -> TokenStream {
+    let (struct_name, struct_fields) =
+        crate::parser::parse_struct_with_named_fields(tokens_input, "ValueList");
+
+    let field_name = struct_fields.named.iter().map(|field| &field.ident);
+    let generated = quote! {
+        impl scylla::frame::value::ValueList for #struct_name {
+            fn serialized(&self) -> scylla::frame::value::SerializedResult {
+                let mut result = scylla::frame::value::SerializedValues::new();
```

Comment:
The values can be initialized with the correct capacity (struct_fields.len())

Reply:
:facepalm: I initially wanted to add capacity, but dropped the idea because `field_name` is an iterator and I don't want to traverse it, but I totally forgot that `struct_fields` is a well defined structure with a length. Thanks